lib/govspeak/header_extractor.rb

```diff
@@ -3,11 +3,45 @@ module Govspeak
 
   class HeaderExtractor < Kramdown::Converter::Base
     def convert(doc)
+      headers = []
+
       doc.root.children.map do |el|
```

Comment:
Because you're now explicitly pushing things into the `headers` array, `each` is more appropriate than `map` here. `map` is used to create an array of the items that the iterator block yields, whereas `each` just executes the block once per item.